pkg/operator/status/status_controller.go

```diff
@@ -149,7 +164,18 @@ func (c StatusSyncer) Sync(ctx context.Context, syncCtx factory.SyncContext) err
 		return nil
 	}
 
-	clusterOperatorObj.Status.RelatedObjects = c.relatedObjects
+	if c.relatedObjectsFunc != nil {
+		ro := c.relatedObjectsFunc()
+		if ro != nil {
+			// append any statically-defined related objects
+			ro = append(ro, c.relatedObjects...)
+			clusterOperatorObj.Status.RelatedObjects = ro
+		} else { // related objects (temporarily) unknown, carry over existing.
+			clusterOperatorObj.Status.RelatedObjects = originalClusterOperatorObj.Status.RelatedObjects
```

Comment:
this has smell. Why not `c.relatedObjects`? You seem to misuse a nil return value as an error case.

Reply:
It's not an error case, it's "I don't know yet". But you're right, that's kind of overloading.

I can change the signature to be more explicit.